### PR TITLE
Force dynamic counter in elementShouldDisplay method to 0

### LINKED ISSUE #55: Bug: Since v4.1.2 valuesAsHtml is not correct for dynamic areas
Since release 4.1.2 the valuesAsHtml() is not correct for dynamic areas. Only the first dynamic area is showing in the html table generated by valuesAsHtml. All other dynamic areas are being ignored. 

Commit https://github.com/neverwoods/validformbuilder/commit/cb14cf1c784a33ca9496d814f44d5e17b8054251 in 4.1.2 is the issue. If I set the `$blnShouldDisplay` to true everything works.

### PR DESCRIPTION
**Pull request by rvanbaalen**
If the condition is met for the first element, it automatically applies to the dynamic clones as well. Therefore we only check dynamic counter 0.

This fixes #55

### CODE REVIEW

File: classes/ValidFormBuilder/ValidForm.php [hunk 1]
```diff
@@ -1847,13 +1847,16 @@ protected function __toJS($strCustomJs = "", $arrInitArguments = array(), $blnRa
      * @param $intDynamicCount
      * @return bool|true
      */
-    protected function elementShouldDisplay(Base $objElement, $intDynamicCount = 0)
+    protected function elementShouldDisplay(Base $objElement)
```
Comment (flangfeldt) [inline classes/ValidFormBuilder/ValidForm.php]: I would not change the method signature. There is already a default value of 0 set, just leave as is.

File: classes/ValidFormBuilder/ValidForm.php [hunk 2]
```diff
@@ -1847,13 +1847,16 @@ protected function __toJS($strCustomJs = "", $arrInitArguments = array(), $blnRa
...
     {
         $blnReturn = true;
 
         $objCondition = $objElement->getConditionRecursive("visible");
         if (is_object($objCondition)) {
-            if ($objCondition->isMet($intDynamicCount)) {
+            // If the condition is met for the first element,
+            // it automatically applies to the dynamic clones as well.
+            // Therefore we only check dynamic counter 0
+            if ($objCondition->isMet(0)) {
```
Comment (flangfeldt) [inline classes/ValidFormBuilder/ValidForm.php]: No need to hard code the 0. It's already set via the default value of $intDynamicCount.

File: classes/ValidFormBuilder/ValidForm.php (hunk 1 above)
Comment (rvanbaalen) [inline classes/ValidFormBuilder/ValidForm.php]: But the second parameter is never used?

Comment (flangfeldt) [inline classes/ValidFormBuilder/ValidForm.php]: Yes, you are right. I thought it was used elsewhere.

**Pull request merged**